xilem_core/src/views/map_state.rs

```diff
@@ -0,0 +1,104 @@
+use core::marker::PhantomData;
+
+use crate::{DynMessage, MessageResult, Mut, View, ViewId, ViewPathTracker};
+
+/// A view that "extracts" state from a [`View<ParentState,_,_>`] to [`View<ChildState,_,_>`].
+/// This allows modularization of views based on their state.
+pub struct MapState<ParentState, ChildState, V, F = fn(&mut ParentState) -> &mut ChildState> {
```

Comment:
Previously named `AdaptState` I'm still not sure about the naming, but I *think* `MapState` makes more sense?

Reply:
I was the one who added `AdaptState`, simply because `Adapt` was already a thing so it seemed like a fitting name. `MapState` seems fine as well.